icepyx/core/visualization.py

```diff
@@ -0,0 +1,340 @@
+                        "region:02d}_{release:03d}_{version:02d}.h5"
+                    )
+                else:
+                    format_string = (
+                        "{product:5d}_{datetime:%Y%m%d%H%M%S}_{rgt:04d}{cycle:02d}{"
+                        "region:02d}_{release:03d}_{version:02d}.h5"
+                    )
+                temp = intake.source.utils.reverse_format(
+                    format_string=format_string, resolved_string=fname
+                )
+
+                rgt = temp['rgt']  # Reference Ground Track
+                cycle = temp['cycle']  # Cycle number
+                f_date = str(temp['datetime'].date())  # Datetime
+
+                paras = [rgt, f_date, cycle, bbox_i, self.product]
+                paras_list.append(paras)
+
+        return paras_list
+
+    @backoff.on_exception(backoff.expo,
+                          (requests.exceptions.Timeout,
+                           requests.exceptions.ConnectionError,
+                           requests.exceptions.ChunkedEncodingError),
+                          max_tries=5)
+    def make_request(self, base_url, payload):
+        """Make HTTP request"""
+        return requests.get(base_url, params=payload)
+
+    def request_OA_data(self, paras):
+        """
+        Request data from OpenAltimetry based on API:
+        https://openaltimetry.org/data/swagger-ui/#/
+
+        :param paras: one parameter list for OpenAltimetry API request
+        :return: xarray dataset
+        """
+
+        base_url = 'https://openaltimetry.org/data/api/icesat2/level3a'
+
+        trackId = paras[0]
+        Date = paras[1]
+        cycle = paras[2]
+        bbox = paras[3]
+        product = paras[4]
+
+        # Generate API
+        payload = {'product': product.lower(),
+                   'endDate': Date,
+                   'minx': str(bbox[0]),
+                   'miny': str(bbox[1]),
+                   'maxx': str(bbox[2]),
+                   'maxy': str(bbox[3]),
+                   'trackId': trackId,
+                   'outputFormat': 'json'}  # default return all six beams
+
+        # request OpenAltimetry
+        r = self.make_request(base_url, payload)
+
+        # get elevation data
+        elevation_data = r.json()
+
+        # length of file list
+        file_len = len(elevation_data['data'])
+
+        # file index satisfies acquisition time from data file
+        idx = [elevation_data['data'][i]['date'] == Date for i in np.arange(file_len)]
+
+        # get data we need
+        beam_data = list(compress(elevation_data['data'], idx))
+
+        if not beam_data:
+            return
+
+        data_name = "lat_lon_elev_canopy" if product == "ATL08" else "lat_lon_elev"
+
+        # iterate six beams, sampling rate: 1/20
+        beam_elev = [beam_data[0]['beams'][i][data_name][::20] for i in np.arange(6) if
+                     beam_data[0]['beams'][i][data_name] != []]
```

Comment:
Now I added a `sample_rate` parameter, for ATL06 it's 1/50, for other products (segment length normally about 100 m) I use 1/20 to avoid over-sampling for low latitude regions.